include/picongpu/plugins/transitionRadiation/Calculator.hpp

```diff
@@ -0,0 +1,238 @@
+/* Copyright 2013-2019 Heiko Burau, Rene Widera, Richard Pausch, Finn-Ole Carstens
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "Particle.hpp"
+
+
+namespace picongpu
+{
+namespace transitionRadiation
+{
+    using complex_X = pmacc::math::Complex< float_X >;
+    using complex_64 = pmacc::math::Complex< float_64 >;
+
+    /* Arbitrary margin which is necessary to prevent division by 0 error
+        * created by particles moving in the plane of the foil.
+        */
+    float_X const DIV_BY_ZERO_MINIMUM = 1.e-7;
+
+    /** Calculator class for calculation of transition radiation.
+     * 
+     * @param particleSet transitionRadiation::Particle to compute transition radiation for
+     * @param lookDirection vector with observation direction
+     */
+    class Calculator
+    {
+
+    private:
+        transitionRadiation::Particle const & particle;
+        float3_X const lookDirection;
```

Comment:
Using a const variable here is great. @psychocoderHPC Do you think having it additionally a reference provide a possible speedup too? (regarding registers used)